token/js/src/instructions/internal.ts

```diff
@@ -9,12 +9,28 @@ export function addSigners(
 ): AccountMeta[] {
     if (multiSigners.length) {
         keys.push({ pubkey: ownerOrAuthority, isSigner: false, isWritable: false });
+        
         for (const signer of multiSigners) {
-            keys.push({
-                pubkey: signer instanceof PublicKey ? signer : signer.publicKey,
-                isSigner: true,
-                isWritable: false,
-            });
+            if (signer instanceof PublicKey || signer instanceof Keypair) {
+                keys.push({
+                    pubkey: signer instanceof PublicKey ? signer : signer.publicKey,
```

Comment:
What if we just did this instead?

```suggestion
                    pubkey: 'publicKey' in signer ? signer.publicKey : signer,
```